src/test/scala/scalaz/nio/BufferSuite.scala

```diff
@@ -21,68 +22,61 @@ object BufferSuite extends RTS {
                }
         } yield c1 == c2
         assert(unsafeRun(testProgram))
-      }
+      }, {
+
+        def allocate = Buffer.byte(initialCapacity)
```

Comment:
I put checks in that 'allocate' section - with tests checking that capacity/position/limit are set - these are the side effects of allocation.

For `position <= limit <= capacity` generally I didn't tested, wanted to do it with scalacheck, but you are right that oracle guys have already tested it for us.

Reply:
Now I see what you mean..  

Actually I did it first like you mean `yield position == 0 && capacity == initialCapacity && limit == initialCapacity` but you don't know which condition out of 3 would break the test, so I splitted them to atomic tests.

Note that I don't check the invariant `position <= limit <= capacity` here but the side effects of setting the allocation, like position is 0.

Reply:
yeah it's more capacity = position = limit. I'm a bit concerned about readability of it. but maybe it's only me.